File: python-pr/52/python/code-quality-readability-maintainability/data_processor_26.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def process_data(data):
    if not data:
        return data
    
    processed_data = []
    for item in data:
        if item and isinstance(item, str) and len(item) > 5:
            processed_data.append(item[:5])
        else:
            processed_data.append(item)
    
    return processed_data
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/data_processor_26.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function processData(data) {
    if (!data || !Array.isArray(data)) {
        return data;
    }
    
    return data.map(item => {
        if (item && typeof item === 'string' && item.length > 5) {
            return item.substring(0, 5);
        }
        return item;
    });
}
// {/fact}